### PR TITLE
[snowflake] pythonic resources SnowflakeIOManager 

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Converts the Snowflake IO manager to the pythonic resource system

## How I Tested These Changes
unit tests

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13232** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13232" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13232?utm_source=stack-comment).

**Review by benpankow — APPROVED** [review]
Great! 🚢

**Review by schrockn — DISMISSED** [review]
Question re: Ellipsis on non-optional types

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/snowflake_io_manager.py [hunk 1]
```diff
@@ -143,6 +106,155 @@ def snowflake_io_manager(init_context):
...
+            @asset(
+                ins={"my_table": AssetIn("my_table", metadata={"columns": ["a"]})}
+            )
+            def my_table_a(my_table: pd.DataFrame) -> pd.DataFrame:
+                # my_table will just contain the data from column "a"
+                ...
+
+    """
+
+    database: str = Field(..., description="Name of the database to use.")
```
[the remark this answers is not in the document]
Comment (schrockn) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/snowflake_io_manager.py:166]: Seems like the `...` has no effect with non-Optional types. Frankly I find Pydantic's use of `...` to be pretty annoying in this respect. I would prefer a `required=True` keyword arg and be done with it.

I recommend that we don't use `...` for non-Optional types in our examples, but would like @benpankow's take

[the remark this answers is not in the document]
Comment (benpankow) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/snowflake_io_manager.py:166]: I have been using the ellipses to indicate 'no default value' when using `Field`, but I am OK omitting it. The Pydantic docs seem to slightly prefer including the ellipses but are not internally consistent, it seems valid to not include it which keeps our models simpler.

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/snowflake_io_manager.py [hunk 2]
```diff
@@ -143,6 +106,155 @@ def snowflake_io_manager(init_context):
...
+    account: str = Field(
+        ...,
+        description="Your Snowflake account name. For more details, see  https://bit.ly/2FBL320.",
+    )
+    user: str = Field(..., description="User login name.")
+    schema_: Optional[str] = Field(
+        default=None, alias="schema", description="Name of the schema to use."
+    )  # schema is a reserved word for pydantic
```
[the remark this answers is not in the document]
Comment (benpankow) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/snowflake_io_manager.py:174]: This is a good idea. Put together a tracking doc here: https://www.notion.so/dagster/Pythonic-config-resources-Common-questions-issues-patterns-048718d9c5964df48f1049ed8528c099

**Pull request merged**